### ISSUE TITLE
Challenge 1 : semantic lane segmantation

### ISSUE THREAD

**Issue by bpself** [bpself/challenge1#1]
The goal of this challenge is to create and train a neural network which detects the vehicle's current lane / lane position.

# Predicable steps:

## Find and normalize training dataset

We don't have resources to build our own training datasets, so we are going to use available annotated one.

* KITTI Semantics : http://www.cvlibs.net/datasets/kitti/
* Mapillary Vistas : http://research.mapillary.com/
* Cityscapes : https://www.cityscapes-dataset.com/

![image](https://user-images.githubusercontent.com/33376338/32429826-b56a9c98-c2cc-11e7-93df-0470646f815a.png)

* Labels and raw images should be normalized.
* Label number should be reduced to 3 : road / road sign / not road

## Build an appropriate NN and train

* Keras lookes like a good choice for the project
* As probably we'll have at most a couple of thousands of images, we can train the network on a normal computer
* Optionally, a GPU can be used

## Semantically process the output of the NN

* Transform the labelled frame image to a 2D birds-eye-view perspective
* Find lanes signs and road edges
* Understand the current position of the camera compared to the road (position to the edges, road curvature, etc)

## Make an ent-to-end NN

The approach above is essentially is a CC approach because we are using perspective transform in order to localise the camera position in the physical word. I argue that using a semantic preprocessing can result a more robust CC pipeline.

Once this approach is built and good enough, we can try to generate output data from the CC pipeline and train a new NN which gets the same raw image data, but instead of outputting labelled image data outputs the camera's position data (distance to the road/lane edge, curvature).

The idea is that a neural network can learn the logic behind perspective transformation and positioning.

So instead of the initial:

**RAW IMAGE DATA -> NN -> CC logic -> Postion**

we'll get a

**RAW IMAGE DATA -> NN -> Position**

pipeline.

As a maybe easier/optional challenge we also could try to build a network which outputs a labelled image transformed into 2D bird's eye view:

**RAW IMAGE DATA -> NN -> position of the road from 2D perspective**
